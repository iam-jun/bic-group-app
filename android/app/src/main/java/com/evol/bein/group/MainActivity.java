package com.evol.bein.group;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import com.facebook.react.ReactActivity;

public class MainActivity extends ReactActivity {

  /**
   * Returns the name of the main component registered from JavaScript. This is used to schedule
   * rendering of the component.
   */
  @Override
  protected String getMainComponentName() {
    return "bein_group";
  }

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(null);
  }

  @Override
  public void onNewIntent(Intent intent) {
    super.onNewIntent(intent);
    setIntent(intent);
  }

}
