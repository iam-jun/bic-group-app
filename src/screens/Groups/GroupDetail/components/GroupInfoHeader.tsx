import React from 'react';
import {View, StyleSheet} from 'react-native';
import {useTheme} from 'react-native-paper';

import Image from '~/beinComponents/Image';
import Icon from '~/beinComponents/Icon';
import {scaleSize} from '~/theme/dimension';
import images from '~/resources/images';
import {ITheme} from '~/theme/interfaces';
import Avatar from '~/beinComponents/Avatar';
import ButtonWrapper from '~/beinComponents/Button/ButtonWrapper';
import Text from '~/beinComponents/Text';
import useGroups from '~/hooks/groups';
import {titleCase} from '~/utils/common';

const GroupInfoHeader = () => {
  const theme = useTheme() as ITheme;
  const styles = themeStyles(theme);
  const groupData = useGroups();
  const {groupDetail} = groupData || {};
  const {name, user_count, icon, background_img_url, privacy} =
    groupDetail?.group || {};

  return (
    <View style={styles.coverAndInfoHeader}>
      {/* Cover photo */}
      <Image
        style={styles.cover}
        source={background_img_url || images.img_cover_default}
      />

      {/* Group info header */}
      <View style={styles.infoContainer}>
        <View style={styles.header}>
          <Avatar.UltraLarge source={icon} style={styles.avatar} />
          <View style={styles.nameHeader}>
            <ButtonWrapper
              textProps={{
                variant: 'h5',
              }}>
              {name}
            </ButtonWrapper>

            <View style={styles.groupInfoText}>
              <Icon
                style={styles.iconSmall}
                icon={'iconPrivate'}
                size={14}
                tintColor={theme.colors.iconTint}
              />
              <Text.BodyS useI18n>{!!privacy && titleCase(privacy)}</Text.BodyS>
              <Text.Subtitle> â¬© </Text.Subtitle>
              <Icon
                style={styles.iconSmall}
                icon={'UsersAlt'}
                size={16}
                tintColor={theme.colors.iconTint}
              />
              <Text.BodySM>{user_count}</Text.BodySM>
            </View>
          </View>
          <ButtonWrapper
            style={styles.chatButton}
            onPress={() => alert('go to chat group')}>
            <Icon
              style={styles.iconSmall}
              icon={'iconMessages'}
              size={32}
              tintColor={theme.colors.iconTint}
            />
          </ButtonWrapper>
        </View>
      </View>
    </View>
  );
};

export default GroupInfoHeader;

const themeStyles = (theme: ITheme) => {
  const {spacing, colors} = theme;
  return StyleSheet.create({
    infoContainer: {
      paddingHorizontal: spacing?.padding.large,
      paddingVertical: spacing?.padding.base,
    },
    header: {
      flexDirection: 'row',
      alignItems: 'center',
    },
    avatar: {
      marginRight: spacing?.margin.base,
    },
    cover: {
      width: scaleSize(375),
      height: scaleSize(210),
    },
    iconSmall: {
      marginRight: spacing?.margin.tiny,
    },
    coverAndInfoHeader: {
      backgroundColor: colors.background,
    },
    headerIcons: {
      flexDirection: 'row',
      justifyContent: 'space-between',
      marginHorizontal: spacing?.margin.large,
      marginVertical: spacing?.margin.small,
    },
    chatButton: {
      backgroundColor: colors.bgButtonSecondary,
      padding: spacing.padding.tiny,
      borderRadius: 6,
    },
    groupInfoText: {
      flexDirection: 'row',
      marginTop: spacing.margin.tiny,
    },
    nameHeader: {flex: 1},
  });
};
