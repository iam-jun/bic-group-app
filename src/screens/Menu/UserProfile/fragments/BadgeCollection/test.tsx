import React from 'react';
import {
  View, Dimensions, FlatList, StyleSheet,
} from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import Avatar from '~/baseComponents/Avatar';
import Icon from '~/baseComponents/Icon';
import spacing, { borderRadius } from '~/theme/spacing';

const Grid = ({ data }: {data: any[]}) => {
  const deviceWidth = Dimensions.get('window').width;

  const itemWidth = 48; // Constant item width
  const numColumns = Math.floor((deviceWidth - 10 * 2) / (itemWidth + 6 * 2));
  const rowWidth = numColumns * (itemWidth + 6 * 2);
  const paddingLeft = (deviceWidth - (10 * 2) - rowWidth) / 2;

  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const styles = themeStyles(theme);

  const isSelected = false;

  const renderGridItem = ({ item, index }: any) =>
    // Render each grid item here
    (
      <View
        key={index}
        style={{
          margin: 6,
          borderWidth: isSelected ? 2 : 1,
          borderColor: isSelected ? colors.purple50 : colors.neutral5,
          borderRadius: borderRadius.pill,
        }}
      >
        <Avatar.Medium
          isRounded
          source={{ uri: item?.iconUrl }}
        />
        {isSelected
          ? (
            <View style={styles.iconChose}>
              <Icon size={10} icon="Check" tintColor={colors.white} />
            </View>
          )
          : null}
      </View>
    );

  return (
    <View style={{
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      paddingHorizontal: 10,
      paddingBottom: 10,
    }}
    >

      <View style={{
        // paddingLeft,
      }}
      >
        <FlatList
          data={data}
          renderItem={renderGridItem}
          keyExtractor={(item, index) => index.toString()}
          numColumns={numColumns}
        />
      </View>
    </View>
  );
};

const themeStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      flex: 1,
      paddingTop: spacing.padding.large,
    },
    iconChose: {
      backgroundColor: colors.purple50,
      borderRadius: borderRadius.pill,
      height: 20,
      width: 20,
      alignItems: 'center',
      justifyContent: 'center',
      position: 'absolute',
      zIndex: 1,
      right: 0,
    },
  });
};

export default Grid;
