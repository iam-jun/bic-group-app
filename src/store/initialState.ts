import {appInitState} from '~/store/app/reducer';
import {modalInitState} from '~/store/modal/reducer';
import {authInitState} from '~/screens/Auth/redux/reducer';
import {postInitState} from '~/screens/Post/redux/reducer';
import {groupInitState} from '~/screens/Groups/redux/reducer';
import {homeInitState} from '~/screens/Home/redux/reducer';
import {notiInitState} from '~/screens/Notification/redux/reducer';
import {menuInitState} from '~/screens/Menu/redux/reducer';
import {noInternetInitState} from '~/screens/NoInternet/redux/reducer';
import {mentionInputInitState} from '~/beinComponents/inputs/MentionInput/redux/reducer';

const initialState = {
  app: appInitState,
  modal: modalInitState,
  auth: authInitState,
  post: postInitState,
  groups: groupInitState,
  home: homeInitState,
  notifications: notiInitState,
  menu: menuInitState,
  noInternet: noInternetInitState,
  mentionInput: mentionInputInitState,
};

export default initialState;
