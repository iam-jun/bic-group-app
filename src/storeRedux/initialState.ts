import { appInitState } from '~/storeRedux/app/reducer';
import { modalInitState } from '~/storeRedux/modal/reducer';
import { authInitState } from '~/storeRedux/auth/reducer';
import { postInitState } from '~/storeRedux/post/reducer';
import { groupInitState } from '~/storeRedux/groups/reducer';
import { homeInitState } from '~/storeRedux/home/reducer';
import { notiInitState } from '~/storeRedux/notification/reducer';
import { menuInitState } from '~/storeRedux/menu/reducer';
import { noInternetInitState } from '~/storeRedux/network/reducer';
import { mentionInputInitState } from '~/beinComponents/inputs/MentionInput/redux/reducer';
import { giphyInitState } from './giphy/reducer';

const initialState = {
  app: appInitState,
  modal: modalInitState,
  auth: authInitState,
  post: postInitState,
  groups: groupInitState,
  home: homeInitState,
  notifications: notiInitState,
  menu: menuInitState,
  noInternet: noInternetInitState,
  mentionInput: mentionInputInitState,
  giphy: giphyInitState,
};

export default initialState;
