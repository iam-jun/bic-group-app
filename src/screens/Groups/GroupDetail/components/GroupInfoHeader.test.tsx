import React from 'react';

import {renderWithRedux, createTestStore, fireEvent} from '~/test/testUtils';
import GroupInfoHeader from './GroupInfoHeader';
import initialState from '~/store/initialState';
import groupJoinStatus from '~/constants/groupJoinStatus';
import {groupDetailData} from '~/test/mock_data/group';

describe('GroupInfoHeader component', () => {
  it('renders correctly', () => {
    const rendered = renderWithRedux(<GroupInfoHeader />);
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it('renders button Join correctly', () => {
    const {queryByTestId} = renderWithRedux(<GroupInfoHeader />);

    // only Join button is available
    const joinButton = queryByTestId('group_info_header.join');
    expect(joinButton).not.toBeNull();
    // @ts-ignore
    fireEvent.press(joinButton);
    const cancelButton = queryByTestId('group_info_header.cancel');
    expect(cancelButton).toBeNull();
  });

  it('renders button Cancel Request correctly', () => {
    const state = {...initialState};
    state.groups.groupDetail.join_status = groupJoinStatus.requested;
    const store = createTestStore(state);
    const {queryByTestId} = renderWithRedux(<GroupInfoHeader />, store);

    //  only Cancel button is available
    const joinButton = queryByTestId('group_info_header.join');
    expect(joinButton).toBeNull();
    const cancelButton = queryByTestId('group_info_header.cancel');
    expect(cancelButton).not.toBeNull();
    // @ts-ignore
    fireEvent.press(cancelButton);
  });

  it('should hide Join and Cancel buttons when user is a group member correctly', () => {
    const state = {...initialState};
    state.groups.groupDetail.join_status = groupJoinStatus.member;
    const store = createTestStore(state);
    const {queryByTestId} = renderWithRedux(<GroupInfoHeader />, store);

    // Join and Cancel buttons are not available
    const joinButton = queryByTestId('group_info_header.join');
    expect(joinButton).toBeNull();
    const cancelButton = queryByTestId('group_info_header.cancel');
    expect(cancelButton).toBeNull();
  });

  it('should render group name correctly', () => {
    const state = {...initialState};
    state.groups.groupDetail = {...groupDetailData};
    const store = createTestStore(state);
    const wrapper = renderWithRedux(<GroupInfoHeader />, store);
    const name = wrapper.getByTestId('group_info_header.name');
    expect(name.props.children).toBe(groupDetailData.group.name);
  });

  it('should render group privacy correctly', () => {
    const state = {...initialState};
    state.groups.groupDetail = {...groupDetailData};
    const store = createTestStore(state);
    const wrapper = renderWithRedux(<GroupInfoHeader />, store);
    const privacy = wrapper.getByTestId('group_info_header.privacy');
    expect(privacy.props.children).toBe('Public Group');
  });

  it('should render group member count correctly', () => {
    const state = {...initialState};
    state.groups.groupDetail = {...groupDetailData};
    const store = createTestStore(state);
    const wrapper = renderWithRedux(<GroupInfoHeader />, store);
    const member = wrapper.getByTestId('group_info_header.member_count');
    expect(member.props.children).toBe(groupDetailData.group.user_count);
  });

  it('should render cover image', () => {
    const state = {...initialState};
    state.groups.groupDetail = {...groupDetailData};
    const store = createTestStore(state);
    const wrapper = renderWithRedux(<GroupInfoHeader />, store);
    const cover = wrapper.getByTestId('group_info_header.cover');
    expect(cover).toBeDefined();
  });

  it('should render avatar image', () => {
    const state = {...initialState};
    state.groups.groupDetail = {...groupDetailData};
    const store = createTestStore(state);
    const wrapper = renderWithRedux(<GroupInfoHeader />, store);
    const avatar = wrapper.getByTestId('avatar');
    expect(avatar).toBeDefined();
  });
});
