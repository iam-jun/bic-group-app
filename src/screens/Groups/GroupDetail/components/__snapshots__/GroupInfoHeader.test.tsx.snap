// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupInfoHeader component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
    }
  }
  testID="group_info_header"
>
  <View
    onLayout={[Function]}
    testID="group_info_header.cover"
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "position": "relative",
        }
      }
    >
      <View
        collapsable={false}
        nativeID="animatedComponent"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#f9f9f9",
              "height": 210,
              "justifyContent": "center",
              "width": "100%",
            }
          }
          testID="Image__placeholder"
        />
      </View>
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "height": 210,
              "width": "100%",
            },
          ]
        }
      >
        <FastImageView
          onFastImageError={[Function]}
          onFastImageLoadEnd={[Function]}
          resizeMode="cover"
          source={
            Object {
              "testUri": "../../../assets/images/cover_default.png",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "paddingHorizontal": 16,
        "paddingVertical": 12,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "borderRadius": 6,
            "height": 48,
            "marginRight": 16,
            "width": 48,
          }
        }
        testID="avatar"
      >
        <View
          style={
            Object {
              "backgroundColor": "#B2BDCD",
              "borderRadius": 6,
              "height": 48,
              "width": 48,
            }
          }
          testID="avatar_container"
        >
          <View
            onLayout={[Function]}
            style={
              Object {
                "backgroundColor": "transparent",
                "position": "relative",
              }
            }
          >
            <View
              collapsable={false}
              nativeID="animatedComponent"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#f9f9f9",
                    "borderRadius": 6,
                    "height": 48,
                    "justifyContent": "center",
                    "width": 48,
                  }
                }
                testID="Image__placeholder"
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 6,
                    "height": 48,
                    "width": 48,
                  },
                ]
              }
            >
              <FastImageView
                onFastImageError={[Function]}
                onFastImageLoadEnd={[Function]}
                resizeMode="cover"
                source={null}
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
                testID="avatar.image"
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={Object {}}
        >
          <Text
            style={
              Object {
                "color": "#2F3945",
                "fontFamily": "OpenSans-SemiBold",
                "fontSize": 20,
              }
            }
            testID="group_info_header.name"
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <View
              accessibilityState={
                Object {
                  "disabled": true,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={false}
              hitSlop={
                Object {
                  "bottom": 10,
                  "left": 10,
                  "right": 10,
                  "top": 10,
                }
              }
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    },
                    Object {
                      "height": 16,
                      "marginRight": 4,
                    },
                    Object {
                      "backgroundColor": undefined,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      undefined,
                      undefined,
                      undefined,
                    ]
                  }
                >
                  <View
                    className="icon-wrapper"
                  >
                    <View
                      onLayout={[Function]}
                      style={
                        Object {
                          "backgroundColor": "transparent",
                          "position": "relative",
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        nativeID="animatedComponent"
                        style={
                          Object {
                            "bottom": 0,
                            "left": 0,
                            "opacity": 1,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#f9f9f9",
                              "height": 16,
                              "justifyContent": "center",
                              "width": 16,
                            }
                          }
                          testID="Image__placeholder"
                        />
                      </View>
                      <View
                        style={
                          Array [
                            Object {
                              "overflow": "hidden",
                            },
                            Object {
                              "height": 16,
                              "width": 16,
                            },
                          ]
                        }
                      >
                        <FastImageView
                          onFastImageError={[Function]}
                          onFastImageLoadEnd={[Function]}
                          resizeMode="cover"
                          size={16}
                          source={null}
                          style={
                            Object {
                              "bottom": 0,
                              "left": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            }
                          }
                        />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <Text
              style={
                Object {
                  "color": "#2F3945",
                  "fontFamily": "OpenSans-Regular",
                  "fontSize": 12,
                }
              }
              testID="group_info_header.privacy"
            />
            <Text
              style={
                Object {
                  "color": "#2F3945",
                  "fontFamily": "OpenSans-Regular",
                  "fontSize": 12,
                }
              }
            >
               • 
            </Text>
            <Text
              style={
                Object {
                  "color": "#2F3945",
                  "fontFamily": "OpenSans-SemiBold",
                  "fontSize": 14,
                }
              }
              testID="group_info_header.member_count"
            />
            <Text
              style={
                Object {
                  "color": "#2F3945",
                  "fontFamily": "OpenSans-Regular",
                  "fontSize": 12,
                }
              }
            >
               members
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "backgroundColor": "#421187",
            "borderRadius": 6,
            "marginLeft": 16,
            "padding": 8,
          }
        }
        testID="group_info_header.join"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
          testID="button_wrapper.content"
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "OpenSans-SemiBold",
                "fontSize": 16,
                "textAlign": "center",
              }
            }
          >
            Join
          </Text>
          <View
            accessibilityState={
              Object {
                "disabled": true,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            hitSlop={
              Object {
                "bottom": 10,
                "left": 10,
                "right": 10,
                "top": 10,
              }
            }
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  Object {
                    "marginHorizontal": 8,
                  },
                  Object {
                    "backgroundColor": undefined,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    undefined,
                    undefined,
                    undefined,
                  ]
                }
              >
                <View
                  className="icon-wrapper"
                  testID="button_wrapper.icon"
                >
                  <RNSVGSvgView
                    align="xMidYMid"
                    bbHeight={20}
                    bbWidth={20}
                    fill="#FFFFFF"
                    focusable={false}
                    height={20}
                    meetOrSlice={0}
                    minX={0}
                    minY={0}
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "borderWidth": 0,
                        },
                        Object {},
                        Object {
                          "flex": 0,
                          "height": 20,
                          "width": 20,
                        },
                      ]
                    }
                    vbHeight={24}
                    vbWidth={24}
                    width={20}
                  >
                    <RNSVGGroup
                      fill={4294967295}
                      propList={
                        Array [
                          "fill",
                        ]
                      }
                    >
                      <RNSVGPath
                        d="M19,11H13V5a1,1,0,0,0-2,0v6H5a1,1,0,0,0,0,2h6v6a1,1,0,0,0,2,0V13h6a1,1,0,0,0,0-2Z"
                      />
                    </RNSVGGroup>
                  </RNSVGSvgView>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
